SalesforceAnalytics.podspec

```diff
@@ -8,12 +8,14 @@ Pod::Spec.new do |s|
   s.license      = { :type => "Salesforce.com Mobile SDK License", :file => "LICENSE.md" }
   s.author       = { "Bharath Hariharan" => "bhariharan@salesforce.com" }
 
-  s.platform     = :ios, "16.0"
+  s.platforms    =  { :ios => "16.0", :visionos => "2.0" }
 
   s.source       = { :git => "https://github.com/forcedotcom/SalesforceMobileSDK-iOS.git",
                      :tag => "v#{s.version}" }
 
-  s.frameworks   = 'CoreTelephony'
```

Comment:
Background on this: 
- I originally added `s.frameworks = 'CoreTelephony'` because it needed to be explicitly linked for catalyst (old comment [here](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/3363#discussion_r610842486)), though testing catalyst again now without this line works
- VisionOS doesn't support CoreTelephony and including in the podspec causes an error 
- I made the inclusion conditional through the xcconfig, could also potentially take out the line completely because because it was only an issue for catalyst and that doesn't seem to be the case anymore